app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/view/SubscribersListFragment.java

```diff
@@ -31,26 +31,34 @@
     private SubscribersListPresenter presenter;
     private SubscribersAdapter subscribersAdapter;
     private SwipeRefreshLayout swipeLayout;
+    private View viewFragment;
 
     private OnFragmentInteractionListener mListener;
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        subscribersAdapter = new SubscribersAdapter();
+        if (savedInstanceState == null) {
+            setRetainInstance(true);
+            subscribersAdapter = new SubscribersAdapter();
+            presenter = new SubscribersListPresenterImpl(this);
+        }
     }
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
-        View viewFragment = inflater.inflate(R.layout.fragment_subscribers_list, container, false);
-        swipeLayout = (SwipeRefreshLayout) viewFragment.findViewById(R.id.swipe_layout);
-        swipeLayout.setOnRefreshListener(this);
-        presenter = new SubscribersListPresenterImpl(this);
-        RecyclerView recyclerViewSubscribers = (RecyclerView) viewFragment.findViewById(R.id.recycler_view_subscribers);
-        recyclerViewSubscribers.setLayoutManager(new LinearLayoutManager(getActivity()));
-        recyclerViewSubscribers.setAdapter(subscribersAdapter);
-        subscribersAdapter.setClickListener(this);
+        if (savedInstanceState == null) {
```

Comment:
If we use that suggestion, the objects recreates every time that fragment creates new, the attributes instances doesn't hold back, it creates a new